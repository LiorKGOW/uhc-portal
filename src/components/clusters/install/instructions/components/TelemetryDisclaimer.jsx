import React from 'react';

import { Text } from '@patternfly/react-core';

import ExternalLink from '~/components/common/ExternalLink';

import links from '../../../../../common/installLinks.mjs';

const TelemetryDisclaimer = () => (
  <Text component="small">
    Red Hat collects a limited amount of telemetry data. By installing OpenShift Container Platform
    4, you accept our data collection policy.{' '}
    <ExternalLink href={links.TELEMETRY_INFORMATION} noIcon>
      Learn more
    </ExternalLink>{' '}
    about the data we collect.
  </Text>
);

export default TelemetryDisclaimer;
