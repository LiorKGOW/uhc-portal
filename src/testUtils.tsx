import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
