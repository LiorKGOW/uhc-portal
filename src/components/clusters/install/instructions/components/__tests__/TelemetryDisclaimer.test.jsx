import React from 'react';

import { checkAccessibility, render } from '~/testUtils';

import TelemetryDisclaimer from '../TelemetryDisclaimer';

describe('<TelemetryDisclaimer />', () => {
  it('is accessible', async () => {
    const { container } = render(<TelemetryDisclaimer />);

    await checkAccessibility(container);
  });
});
