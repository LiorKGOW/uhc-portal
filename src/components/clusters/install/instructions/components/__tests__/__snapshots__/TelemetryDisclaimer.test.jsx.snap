// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelemetryDisclaimer renders correctly 1`] = `
<Text
  component="small"
>
  Red Hat collects a limited amount of telemetry data. By installing OpenShift Container Platform 4, you accept our data collection policy.
   
  <Text
    component="a"
    href="https://docs.openshift.com/container-platform/4.12/support/remote_health_monitoring/about-remote-health-monitoring.html"
    rel="noreferrer noopener"
    target="_blank"
  >
    Learn more
  </Text>
   
  about the data we collect.
</Text>
`;
